controllers/appointment.controller.js

```diff
@@ -0,0 +1,168 @@
+const objectConverter = require("../utils/objectConverter");
+const HealthTrackRecord = require("../models/healthTrackRecord.model");
+const constants = require("../utils/constants");
+const User = require("../models/user.model");
+const Appointment = require('../models/appointment.model');
+const Hospital = require('../models/hospital.model');
+
+//adding the appointment
+exports.addAppointment = async (req, res) => {
+
+  
+
+    try{
+        //getting the user who is looged in 
+        const user = await User.findOne({userId : req.userId});
+
+        //Getting a doctor id from response body
+        const doctor = await User.findOne({_id : req.body.doctorId});
+
+        //checking is doctor exsit
+        if(doctor == null){
+            return res.status(400).send({
+                message : "Doctor does not exsist please enter valid doctor id"
+            })
+        };
+
+        //Getting a hospital id from response body
+        const hospital = await Hospital.findOne({_id : req.body.hospital_ids});
+
+        //checking is hospital exsit
+        if(hospital == null){
+            return res.status(400).send({
+                message : "Hospital does not exsist please enter valid hospital id"
+            })
+        };
+
+        //checking if the doctor is working in the hospital provided by user
+        if(JSON.stringify(hospital.doctor_ids) == JSON.stringify(doctor.hospitalId)){
```

Comment:
hospital.doctor_ids - doctorId
doctor.hospitalId.  - hospitalId

These two are not comparable values